specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -553,13 +569,20 @@ def verify_cell_kzg_proof_batch(row_commitments_bytes: Sequence[Bytes48],
 ### `construct_vanishing_polynomial`
 
 ```python
-def construct_vanishing_polynomial(missing_cell_ids: Sequence[CellID]) -> Tuple[
-        Sequence[BLSFieldElement],
-        Sequence[BLSFieldElement]]:
+def construct_vanishing_polynomial(missing_cell_ids: Sequence[CellID]) -> Sequence[BLSFieldElement]:
     """
-    Given the cells that are missing from the data, compute the polynomial that vanishes at every point that
+    Given the cells IDs that are missing from the data, compute the polynomial that vanishes at every point that
     corresponds to a missing field element.
+    
+    This method assumes that all of the cells cannot be missing. In this case the vanishing polynomial
+    could be computed as Z(x) = x^n - 1, where `n` is FIELD_ELEMENTS_PER_EXT_BLOB.
+
+    We never encounter this case however because this method is used solely for recovery and recovery only
+    works then at most half of the cells are missing.
```

Comment:
Something feels wrong the latter part of this sentence:

> and recovery only works then at most half of the cells are missing.

Maybe it should be:

> and recovery only works if half of the cells are available.

Reply:
I can switch it around to say "at least" instead of "at most"

Reply:
That would work. But my confusion is with the "works then at" part.

Reply:
It might also be missing an "if" word. "only works then **if** at most".